File: dixit-companion/src/main/java/fr/erban/dxitcompanion/db/player/PlayerWithGamesEntity.java
package fr.erban.dxitcompanion.db.player;

import androidx.room.Embedded;
import androidx.room.Junction;
import androidx.room.Relation;

import java.util.List;

import fr.erban.dxitcompanion.db.crossref.GamePlayerCrossRefEntity;
import fr.erban.dxitcompanion.db.game.GameEntity;

public class PlayerWithGamesEntity {

    @Embedded
    public PlayerEntity player;

    @Relation(
            parentColumn = "name",
            entityColumn = "idGame",
            associateBy = @Junction(GamePlayerCrossRefEntity.class)
    )
    public List<GameEntity> games;
}

File: dixit-companion/src/main/java/fr/erban/dxitcompanion/db/player/PlayerWithGamesEntity.kt
package fr.erban.dxitcompanion.db.player

import androidx.room.Embedded
import androidx.room.Junction
import androidx.room.Relation
import fr.erban.dxitcompanion.db.crossref.GamePlayerCrossRefEntity
import fr.erban.dxitcompanion.db.game.GameEntity

data class PlayerWithGamesEntity(
    @Embedded val player: PlayerEntity,
    @Relation(
        parentColumn = "name",
        entityColumn = "idGame",
        associateBy = Junction(GamePlayerCrossRefEntity::class)
    )
    val games: List<GameEntity>
)
